File: src/service/divide_client.py
#!/usr/bin/env python

from ros_basics_tutorials.srv import DivideTwoInts
from ros_basics_tutorials.srv import DivideTwoIntsRequest
from ros_basics_tutorials.srv import DivideTwoIntsResponse

import rospy
import time
import sys


def divide_two_ints_client(x, y):
    rospy.wait_for_service('divide_two_ints')
    try:
        divide_two_ints = rospy.ServiceProxy('divide_two_ints', DivideTwoInts)
        resp1 = divide_two_ints(x, y)
        return resp1.result
    except rospy.ServiceException as e:
        print("Service call failed: %s" % e)


if __name__ == "__main__":
    if len(sys.argv) == 3:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
    else:
        print("%s [x y]" % sys.argv[0])
        sys.exit(1)
    print("Requesting %s+%s" % (x, y))
    s = divide_two_ints_client(x, y)
    print("%s / %s = %s" % (x, y, s))

File: src/service/divide_client.cpp
#include "ros/ros.h"
#include "ros_basics_tutorials/DivideTwoInts.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "divide_two_ints_client");
    if (argc != 3)
    {
        ROS_INFO("usage: divide_two_ints_client x y");
        return 1;
    }

    ros::NodeHandle n;
    ros::ServiceClient client = n.serviceClient<ros_basics_tutorials::DivideTwoInts>("divide_two_ints");
    ros_basics_tutorials::DivideTwoInts srv;

    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);

    if (client.call(srv))
    {
        ROS_INFO("Result: %lf", (double)srv.response.result);
    }
    else
    {
        ROS_ERROR("Failed to call service divide_two_ints");
        return 1;
    }

    return 0;
}